packages/core/lib/commands/test/run.js

```diff
@@ -68,7 +68,41 @@ module.exports = async function (options) {
     inputFile: file
   });
 
-  if (config.networks[config.network]) {
+  /**
+   * If test network exists and provider, host or url doesn't exist, then internal ganache is started with the user specified
+   * configurations in the config.
+   * If development or test network with provider, host or url exists, then it connects to that network specified in the config.
+   * If no network is specified in the config, then internal ganache is started with default configuration.
+   */
+  const configuredNetwork = config.networks[config.network];
+  const testNetworkDefined = configuredNetwork && config.network == "test";
+  const noProviderHostOrUrlConfigured =
+    !configuredNetwork["provider"] &&
+    !configuredNetwork["host"] &&
+    !configuredNetwork["url"];
```

Comment:
Oops, should verify `configuredNetwork` is not undefined! Sorry about that!
```suggestion
  const noProviderHostOrUrlConfigured =
    configuredNetwork &&
    !configuredNetwork.provider &&
    !configuredNetwork.host &&
    !configuredNetwork.url;
```